cpp/include/cudf/io/text/data_chunk_source.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/cuda_stream_pool.hpp>
+#include <rmm/device_buffer.hpp>
+
+namespace cudf {
+namespace io {
+namespace text {
+
+struct data_chunk {
```

Comment:
This is really just a holder `device_span` and serves no purpose at this point. It would be serving as the `raii` for an `rmm::device_buffer`, but I'm having to workaround some rmm issues for the time being.

Reply:
As you said, I'm not sure what the purpose of the `data_chunk` type is.

Reply:
If we allowed RMM to handle the memory management, rather than using persistent "tickets"/caches in `data_chunk_reader`, then this class would be responsible for owning the rmm buffer, and subsequently destroying it after the end-user was done queueing work for the data chunk. It can just be an `rmm::device_buffer`, though, because it might be a view over an existing `device_buffer`, so there needs to be a layer of indirection, hence this class.

The reason we're using explicit "tickets"/caches right now is because creating short-lived buffers like these in non-default streams in RMM causes undesired synchronizations, preventing all overlap.

Reply:
So basically this class serves as a placeholder for a more general idea.

Reply:
I removed `data_chunk` in favor of `device_span` until it another type become necessary.